odoo/addons/base/models/res_config.py

```diff
@@ -413,14 +413,10 @@ def fields_view_get(self, view_id=None, view_type='form',
         return ret_val
 
     def onchange_module(self, field_value, module_name):
-        ModuleSudo = self.env['ir.module.module'].sudo()
-        modules = ModuleSudo.search(
-            [('name', '=', module_name.replace("module_", '')),
-            ('state', 'in', ['to install', 'installed', 'to upgrade'])])
-
-        if modules and not int(field_value):
-            deps = modules.sudo().downstream_dependencies()
-            dep_names = (deps | modules).mapped('shortdesc')
+        module = self.env['ir.module.module'].sudo()._get(module_name[7:])
```

Comment:
`sudo` seems unnecessary here? `_get` has no security protections.

Reply:
I kept the old behavior where the returned modules records where sudoed, but I'm not sure it's necessary anymore, as only admins are supposed to modify settings.